### PR TITLE
Fix ES debug mode test failing with missing batch

### PR DESCRIPTION
**Pull request by stiepan**
Signed-off-by: Kamil Tokarski <ktokarski@nvidia.com>

## Category:

**Bug fix** (*non-breaking change which fixes an issue*) 

## Description:
Increases the number of input batches fed into the external source to make allowance for `prefetch_queue_depth`, so that the test does not randomly fail with missing batch error.

### Tests
- [ ] Existing tests apply
- [ ] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [x] N/A

### Documentation
- [ ] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [x] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [x] N/A

**REQ IDs**: N/A

**JIRA TASK**: DALI-2621

### CODE REVIEW

File: dali/test/python/test_pipeline_debug.py [hunk 1]
```diff
@@ -187,12 +187,13 @@ def es_pipeline_standard():
...
+    prefetch_queue_depth = 2
     pipe_load = load_images_pipeline()
-    pipe_standard = es_pipeline_standard()
-    pipe_debug = es_pipeline_debug()
+    pipe_standard = es_pipeline_standard(prefetch_queue_depth=prefetch_queue_depth)
+    pipe_debug = es_pipeline_debug(prefetch_queue_depth=prefetch_queue_depth)
     pipe_load.build()
     pipe_debug.build()
-    for _ in range(n_iters):
+    for _ in range(n_iters + prefetch_queue_depth):
```
Comment (JanuszL) [inline dali/test/python/test_pipeline_debug.py:196]: I would add a comment why we need that `+ prefetch_queue_depth` here.

File: dali/test/python/test_pipeline_debug.py [hunk 2]
```diff
@@ -209,12 +210,13 @@ def es_pipeline(source, batch):
...
-    pipe_debug = es_pipeline(source, batch, debug=True)
-    pipe_standard = es_pipeline(source, batch)
+    prefetch_queue_depth = 2
+    pipe_debug = es_pipeline(source, batch, prefetch_queue_depth=prefetch_queue_depth, debug=True)
+    pipe_standard = es_pipeline(source, batch, prefetch_queue_depth=prefetch_queue_depth)
     pipe_debug.build()
     pipe_standard.build()
     if source is None:
-        for _ in range(n_iters):
+        for _ in range(n_iters + prefetch_queue_depth):
```
Comment (JanuszL) [inline dali/test/python/test_pipeline_debug.py:219]: As above.

File: dali/test/python/test_pipeline_debug.py (hunk 1 above)
Comment (stiepan) [reply dali/test/python/test_pipeline_debug.py:198]: done

File: dali/test/python/test_pipeline_debug.py (hunk 2 above)
Comment (stiepan) [reply dali/test/python/test_pipeline_debug.py:223]: done

**Review by JanuszL — APPROVED** [review]

**Comment by stiepan** [comment]
!build

**Review by prak-nv — APPROVED** [review]

**Pull request merged**